From e3298d7a8f733d854d5e3e647ff214ea984e6d03 Mon Sep 17 00:00:00 2001
From: Andrew Walsh <awalsh@redhat.com>
Date: Wed, 15 Jun 2022 19:38:56 -0400
Subject: [PATCH 02/13] Removed/fixed stdarg.h imports

Generally speaking, the stdarg.h import wasn't needed anymore, but in
vdo/kernel/vdoStringUtils.h it still has a dependency on the symbols
contained therein.
---
 uds/memoryAlloc.h           | 3 +++
 uds/permassertInternals.h   | 3 +++
 uds/stringUtils.h           | 3 +++
 vdo/base/vio.h              | 3 +++
 vdo/kernel/logger.h         | 4 +++-
 vdo/kernel/vdoStringUtils.h | 7 ++++++-
 6 files changed, 21 insertions(+), 2 deletions(-)

diff --git a/uds/memoryAlloc.h b/uds/memoryAlloc.h
index ea18fa5..8a1427c 100644
--- a/uds/memoryAlloc.h
+++ b/uds/memoryAlloc.h
@@ -22,7 +22,10 @@
 #ifndef MEMORY_ALLOC_H
 #define MEMORY_ALLOC_H 1
 
+#include <linux/version.h>
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(5,15,0)
 #include <stdarg.h>
+#endif
 
 #include "compiler.h"
 #include "cpu.h"
diff --git a/uds/permassertInternals.h b/uds/permassertInternals.h
index 35f72d5..2d360e1 100644
--- a/uds/permassertInternals.h
+++ b/uds/permassertInternals.h
@@ -22,7 +22,10 @@
 #ifndef PERMASSERT_INTERNALS_H
 #define PERMASSERT_INTERNALS_H
 
+#include <linux/version.h>
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(5,15,0)
 #include <stdarg.h>
+#endif
 
 #ifdef __cplusplus
 extern "C" {
diff --git a/uds/stringUtils.h b/uds/stringUtils.h
index 925cb36..17117b7 100644
--- a/uds/stringUtils.h
+++ b/uds/stringUtils.h
@@ -22,7 +22,10 @@
 #ifndef STRING_UTILS_H
 #define STRING_UTILS_H
 
+#include <linux/version.h>
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(5,15,0)
 #include <stdarg.h>
+#endif
 #ifdef __KERNEL__
 #include <linux/kernel.h>
 #include <linux/string.h>
diff --git a/vdo/base/vio.h b/vdo/base/vio.h
index de38144..c4c0884 100644
--- a/vdo/base/vio.h
+++ b/vdo/base/vio.h
@@ -22,7 +22,10 @@
 #ifndef VIO_H
 #define VIO_H
 
+#include <linux/version.h>
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(5,15,0)
 #include <stdarg.h>
+#endif
 
 #include "completion.h"
 #include "trace.h"
diff --git a/vdo/kernel/logger.h b/vdo/kernel/logger.h
index 91bacd5..60a68ee 100644
--- a/vdo/kernel/logger.h
+++ b/vdo/kernel/logger.h
@@ -22,10 +22,12 @@
 #ifndef LOGGER_H
 #define LOGGER_H 1
 
-#include <stdarg.h>
 #include <linux/sched.h>
 #include <linux/types.h>
 #include <linux/version.h>
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(5,15,0)
+#include <stdarg.h>
+#endif
 
 #define LOG_EMERG       0       /* system is unusable */
 #define LOG_ALERT       1       /* action must be taken immediately */
diff --git a/vdo/kernel/vdoStringUtils.h b/vdo/kernel/vdoStringUtils.h
index c9adf87..6b91f47 100644
--- a/vdo/kernel/vdoStringUtils.h
+++ b/vdo/kernel/vdoStringUtils.h
@@ -22,8 +22,13 @@
 #ifndef VDO_STRING_UTILS_H
 #define VDO_STRING_UTILS_H
 
-#include <stdarg.h>
 #include <linux/types.h>
+#include <linux/version.h>
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(5,15,0)
+#include <stdarg.h>
+#else
+#include <linux/stdarg.h>
+#endif
 
 /**
  * Helper to append a string to a buffer.
-- 
2.35.3

